services/httppost/service.go

```diff
@@ -27,25 +27,55 @@ func NewService(c Configs, l *log.Logger) *Service {
 	return s
 }
 
+// Only one of name and url should be non-empty
+type Endpoint struct {
+	s      *Service
+	name   string
+	url    string
+	logger *log.Logger
+}
+
+func (e *Endpoint) NewHTTPRequest(body io.Reader) (req *http.Request, err error) {
+	if e.url != "" {
+		req, err = http.NewRequest("POST", e.url, body)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create POST request: %v", err)
+		}
+		return
+	}
+	c, ok := e.s.endpoint(e.name)
```

Comment:
This is the bit I missed before, we reread the config on every NewHTTPRequest, which is why it is working.